docs/cli-usage.md

```diff
@@ -0,0 +1,96 @@
+## Using Kubernetes from the command line
```

Comment:
And command-line shorthand: http://docs.aws.amazon.com/cli/latest/userguide/shorthand-syntax.html

I could also imagine supporting json path syntax.

I'd like to avoid hardcoding lots of custom command-line flags in kubectl -- they're hard to maintain, exacerbate client compatibility problems, gets complicated when supporting multiple API versions, and doesn't work for API plugins.

If we wanted custom encoding of fields as command-line arguments (not cobra pflags), one approach could be to annotate the fields with these custom flag names and query the API to pull down that metadata. With the object schema in hand, we could automatically infer nesting depth from an ordered sequence of fields.

Definitely the easiest way to support objects embedded on the command line would be to just take actual json/yaml from a command-line argument (e.g., `-j`/`-y`). As an example, here's what a simple v1beta1 replicationController would look like (with implicitly quoted keys but explicitly quoted values, and implicit apiVersion and Kind):
```
kubectl create rc -j '{id:"redismaster",desiredState:{replicas:1,replicaSelector:{simpleservice:"redismaster"},podTemplate:{desiredState:{manifest:{containers:[name:"redismaster",image:"dockerfile/redis"]}},labels:{simpleservice:"redismaster"}}},labels:{simpleservice:"redismaster"}}'
``` 
v1beta3 will cut down on the nesting a little. Some intelligent defaulting would go a long way (e.g., inferring the selector and replicationController labels from the pod template labels).